app.rb

```diff
@@ -5,3 +5,122 @@
 require_relative './modules/label'
 require_relative './modules/music_album'
 require_relative './modules/source'
+require 'date'
+class App
+  def initialize(things)
+    @things = things
+  end
+
+  def create_book
+    puts 'to create book please enter the following information:'
+    print "Enter book's published date(format: YYYY/MM/DD): "
+    published_date = Date.parse(gets.chomp)
+    print "Enter book's publisher: "
+    publisher = gets.chomp
+    print "Enter book's cover state: "
+    cover_state = gets.chomp
+    book = Book.new(published_date, publisher, cover_state)
+    book.add_author(create_author)
+    book.add_label(create_label)
+    book.add_source(create_source)
+    book.add_genre(create_genre)
+    @things.add_book(book)
+    puts 'Book added successfully!'
+  end
+
+  def list_books
+    puts 'Avaible books are:'
+    @things.books.each do |book|
+      puts "This book is in #{book.genre.name} genre and published at #{book.publish_date} and  #{book.cover_state} cover state."
+    end
+  end
+
+  def create_game
+    puts 'to create game please enter the following information:'
+    print "Enter game's published date(format: YYYY/MM/DD): "
+    published_date = Date.parse(gets.chomp)
+    print 'Enter game multiplayer mode:[true/false] '
+    multiplayer = gets.chomp.to_s
+    multiplayer = multiplayer == 'true'
+    print "Enter game's last played date(format: YYYY/MM/DD): "
+    last_played_at = Date.parse(gets.chomp)
+    game = Game.new(published_date, multiplayer, last_played_at)
+    game.add_author(create_author)
+    game.add_label(create_label)
+    game.add_source(create_source)
+    game.add_genre(create_genre)
+    @things.add_game(game)
+    puts 'Game added successfully!'
+  end
+
+  def list_games
+    puts 'Avaible games are:'
+    @things.games.each do |game|
+      puts "This game is in #{game.genre.name} genre and published at #{game.publish_date} and multiplayer mode  #{game.multiplayer} ."
+    end
```

Comment:
- [optional] It would be great if you add a counter for better appearance like:

```
puts 'Avaible games are:'
count = 0

@things.games.each do |game|
      puts "#{count}) [Game] This game is in #{game.genre.name} genre and published at #{game.publish_date} and multiplayer mode  #{game.multiplayer} ."
    count += 1
end 
```